### ISSUE TITLE
NextDNS is not Anycast but Geo-Steered (DoH and DoT)

### ISSUE THREAD

**Issue by ignoramous** [privacyguides/privacyguides.org#1898]
## Description

AFA I can tell, DoT and DoH running at `<any>.dns.nextdns.io` is geo load-balanced and not Anycast (they do it for good reasons, I'd presume, namely for their purported "ultra-low latency" feature). As it stands, they don't meet PrivacyGuides' criteria (a criteria that I don't agree with, but that's a separate discussion).

To verify geo-steering, try fetching DNS records from different locations.

Here's what I see:
```
# From Madras, India:
dns.nextdns.io	canonical name = steering.nextdns.io.
Address: 139.84.131.131 (Vultr)
Address: 68.183.91.243 (Digital Ocean)

# From Washington DC, US:
dns.nextdns.io	canonical name = steering.nextdns.io
Address: 199.119.65.94 (Misaka)
Address: 5.161.43.197 (Hetzner)
```

Though, there's no way to ascertain whether a unicast IP is indeed Anycast, it is easy to see with `ping`s and `traceroute`s that there's a chance it isn't. Because in a typical deployment, Anycast IPs don't have super high pings from different parts of the globe.

```
# Ping Misaka addr form Madras, India
ping  199.119.65.94
PING 199.119.65.94 (199.119.65.94) 56(84) bytes of data.
64 bytes from 199.119.65.94: icmp_seq=1 ttl=42 time=482 ms
64 bytes from 199.119.65.94: icmp_seq=3 ttl=42 time=421 ms
64 bytes from 199.119.65.94: icmp_seq=4 ttl=42 time=414 ms

# Ping Hetzner addr from Madras, India
ping 5.161.43.197
PING 5.161.43.197 (5.161.43.197) 56(84) bytes of data.
64 bytes from 5.161.43.197: icmp_seq=1 ttl=41 time=426 ms
64 bytes from 5.161.43.197: icmp_seq=2 ttl=41 time=449 ms
64 bytes from 5.161.43.197: icmp_seq=3 ttl=41 time=464 ms

---

# Ping Vultr from Washington DC, US:
ping 139.84.131.131 
PING 139.84.131.131 (139.84.131.131): 56 data bytes
64 bytes from 139.84.131.131: seq=0 ttl=48 time=243.380 ms
64 bytes from 139.84.131.131: seq=1 ttl=48 time=243.248 ms
64 bytes from 139.84.131.131: seq=1 ttl=48 time=250.116 ms

# (no response!) Ping Digital Ocean from Washington DC, US:
ping 68.183.91.243
PING 68.183.91.243 (68.183.91.243): 56 data bytes
^C
--- 68.183.91.243 ping statistics ---
15 packets transmitted, 0 packets received, 100% packet loss
``` 

**URL of affected page:** https://www.privacyguides.org/dns/#recommended-providers

For me, personally, geo-steering is fine, but I don't know why PrivacyGuides has a criteria for Anycast here.

---

Disclosure: I run an Anycast DoH and DoT resolver myself but this post isn't a shade on NextDNS (in fact, they're pretty good at what they do).

**Comment by lunik1** [comment]
It is not very well advertised, but `anycast.dns.nextdns.io` exists and is not geo-steered afaict, but it seems you have a better range of locations from which to confirm this than I do!

**Comment by ignoramous** [comment]
> Since NextDNS does have anycast support, I'll go ahead and close this issue.

The default / advertised endpoint (`dns.nextdns.io` and its variants) aren't Anycast.

**Comment by mfwmyfacewhen** [comment]
Well the requirement is only that it's supported at all.

**Comment by ignoramous** [comment]
> Well the requirement is only that it's supported at all.

From PrivacyGuides.org:

> > Must support DNSSEC
> > Must have anycast support
> > QNAME Minimization
> > Allow for ECS to be disable

If one implements a DNS resolver which supports these (say, on `some.obscure.sub.domain.tld`) but not on the default endpoint (say, on `dns.domain.tld`), they all good?

**Comment by dngray** [comment]
I suppose we could remove the requirement. The reason we added it in the first place was to filter out `insert 100 small DNS forwarders` being recommended constantly. Not that we have anything against small DNS servers, but we think that fundamentally when they have no privacy policy, or can't really be trusted to follow the one they have, there is not any point in adding them.

I think rather than trying to make a strict criteria for this section, we might just use our discretion.

**Comment by ignoramous** [comment]
> but we think that fundamentally when they have no privacy policy

This assumes the larger dns providers have apt privacy policy. Since we are talking about NextDNS here, it doesn't take much long to know their privacy policy is rather lacking. They don't mention Cloudflare, GCP, DigitalOcean, AWS, Vercel or any other infrastructure provider they use. You folks need to be careful going by popularity too, just as you're careful with small scale resolvers ran by enthusiasts.

**Comment by dngray** [comment]
> Since we are talking about NextDNS here, it doesn't take much long to know their privacy policy is rather lacking. They don't mention Cloudflare, GCP, DigitalOcean, AWS, Vercel or any other infrastructure provider they use. You folks need to be careful going by popularity too, just as you're careful with small scale resolvers ran by enthusiasts.

They don't but then we don't consider DNS anonymous anyway. Most businesses which have sort of public front or service will rely on other providers for DDoS protection or other services.

> You folks need to be careful going by popularity too

It's not really about that either, its more that it's not someone's server in someone's basement that could be set up in whatever crappy way possible. We want to get a feel that it's actually set up by someone resourced enough to make sure everything is best practice.

That is difficult to determine from our position, but we do try our best.

**Comment by ignoramous** [comment]
Makes sense. Thanks! This is work is hard yet voluntary, so I don't envy you folks.

> Most businesses which have sort of public front or service will rely on other providers for DDoS protection or other services.

That wasn't my point. It was more around not being transparent at all with privacy policy. Besides, NextDNS doesn't use GCP / DO / AWS for DDoS. It uses them for logs and analytics (from what I can tell). This is problematic because folks use services that replicate pi-hole like service mostly to de-google / de-amazon / whatever. They look at PrivacyGuides recommendation, then glance at privacy policy and think that's the entire story.

PrivacyTools (when it was run by the same community as PrivacyGuides) was ruthless about this kind of stuff.

**Comment by dngray** [comment]
> Besides, NextDNS doesn't use GCP / DO / AWS for DDoS. It uses them for logs and analytics (from what I can tell). This is problematic because folks use services that replicate pi-hole like service mostly to de-google / de-amazon / whatever.

That does seem like content for an appropriate blog post, which we now have tags for things like this https://blog.privacyguides.org/tags/ we do take contributed posts as long as they are high quality.

I had some plans to investigate Freedombox further, to see what that could do in regard to self hosting some things.

> PrivacyTools (when it was run by the same community as PrivacyGuides) was ruthless about this kind of stuff.

The site at that time also did lack a clear threat model, in some places it was all about "the EyEs" in other places, not so much. I think now we've got a more comprehensive and "leveled" threat model.

TLDR is though use the right tool for the right job:

- Trying to hide your IP? Tor or a VPN
- Want nobody to see your data? Use audited E2EE.
